### PR TITLE
Fix status response handling in write handler

### PR DESCRIPTION
**Pull request by yunhanw-google**
#### Problem
Fix status response handler in ReadHandler
This PR is cut from https://github.com/project-chip/connectedhomeip/pull/19356

#### Change overview
When receiving unknown message in write handler, write handler would send status response with invalid action.

#### Testing
Write client sends out the chunked write request and receive response, then inject the unknown report message in write handler side, write handler would send status response with invalid action.

### CODE REVIEW

File: src/app/WriteHandler.cpp [hunk 1]
```diff
@@ -118,7 +118,13 @@ CHIP_ERROR WriteHandler::OnMessageReceived(Messaging::ExchangeContext * apExchan
                        "OnMessageReceived should not be called on GroupExchangeContext");
     if (!aPayloadHeader.HasMessageType(Protocols::InteractionModel::MsgType::WriteRequest))
     {
+        if (aPayloadHeader.HasMessageType(Protocols::InteractionModel::MsgType::StatusResponse))
+        {
+            CHIP_ERROR statusError = CHIP_NO_ERROR;
+            StatusResponse::ProcessStatusResponse(std::move(aPayload), statusError);
```
Comment (bzbarsky-apple) [inline src/app/WriteHandler.cpp:124]: What's the point of this block?  `statusError` never escapes....  If this is just for the logging, please document that.

File: src/app/tests/TestWriteInteraction.cpp [hunk 2]
```diff
@@ -543,6 +551,74 @@ void TestWriteInteraction::TestWriteRoundtrip(nlTestSuite * apSuite, void * apCo
     engine->Shutdown();
 }
 
+// This test is to create Chunked write requests, we drop the message since the 3rd message, then write hanlder received unknown
+// report message and sends out the status report with invalid action.
```
Comment (bzbarsky-apple) [inline src/app/tests/TestWriteInteraction.cpp:555]:
```suggestion
// This test creates a chunked write request, we drop the second write chunk message, then write handler receives unknown
// report message and sends out a status report with invalid action.
```

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/WriteHandler.cpp (hunk 1 above)
Comment (yunhanw-google) [reply src/app/WriteHandler.cpp:124]: yes, it is for logging purpose!

File: src/app/tests/TestWriteInteraction.cpp (hunk 2 above)
Comment (yunhanw-google) [reply src/app/tests/TestWriteInteraction.cpp]: done, thanks

**Review by yufengwangca — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@yunhanw-google This is failing the nRF Connect unit tests, with this error:
```
2022-08-06T06:29:13.5184322Z 2022-08-06 06:28:12 INFO    ../../../../src/app/tests/TestWriteInteraction.cpp:588: assertion failed: "InteractionModelEngine::GetInstance()->GetNumActiveWriteHandlers() == 1"
```

I am guessing the PacketBuffers there are bigger, and we end up not chunking....  We should probably reserve an amount of space dependent on the packet buffer available space, not a hardcoded 900 bytes.

**Pull request merged**